src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -276,32 +297,37 @@ public void visitToken(DetailAST ast) {
      * Checks whether a method has a javadoc comment.
      *
      * @param methodDef method definition token.
+     * @param requiredJavadocPhrase required javadoc comment pattern.
      * @return true if a method has a javadoc comment.
      */
-    private static boolean hasJavadocComment(DetailAST methodDef) {
-        return hasJavadocCommentOnToken(methodDef, TokenTypes.MODIFIERS)
-                || hasJavadocCommentOnToken(methodDef, TokenTypes.TYPE);
+    private static boolean hasJavadocComment(DetailAST methodDef, Pattern requiredJavadocPhrase) {
```

Comment:
I am not 100% sure, but I think it is reasonable to make method `non static` and use `requiredJavadocPhrase` as class field. 
I think it was made `static` because previous implementation was not using `Check` properties 
@rnveach please correct me if I am wrong